File: MicroTube/Controllers/Authentication/DTO/VideoUploadProgressDTO.cs
using MicroTube.Data.Models;

namespace MicroTube.Controllers.Authentication.DTO
{
	public class VideoUploadProgressDTO
	{
		public string Id { get; set; }
		public string Title { get; set; }
		public string? Description { get; set; }
		public VideoUploadStatus Status { get; set; }
		public VideoUploadProgressDTO(string id, VideoUploadStatus status, string title, string? description)
		{
			Id = id;
			Status = status;
			Title = title;
			Description = description;
		}
	}
}

File: MicroTube/ClientApp/src/app/data/DTO/VideoUploadProgressDTO.ts
export interface VideoUploadProgressDTO
{
  id: string,
  title: string;
  description: string | null;
  status: string;
  message: string | null;
  lengthSeconds: number | null,
  lengthHuman: string | null;
  timestamp: string;
  timestampHuman: string | null;
}
